PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt

```diff
@@ -85,10 +107,63 @@ class PayPalWebCheckoutClient internal constructor(
      *
      * @param request [PayPalWebCheckoutRequest] for requesting an order approval
      */
-    suspend fun start(
+    @Deprecated(
+        message = "Use startAsync(activity, request) or start(activity, request, callback) instead.",
+        replaceWith = ReplaceWith("startAsync(activity, request)")
+    )
+    fun start(
+        activity: Activity,
+        request: PayPalWebCheckoutRequest
+    ): PayPalPresentAuthChallengeResult {
+        checkoutOrderId = request.orderId
+        analytics.notify(CheckoutEvent.STARTED, checkoutOrderId)
+
+        val launchUri = buildPayPalCheckoutUri(
+            orderId = request.orderId,
+            funding = request.fundingSource,
+            redirectUrl = request.appLinkUrl ?: redirectUriPayPalCheckout
+        )
+
+        val result = payPalWebLauncher.launchWithUrl(
+            activity = activity,
+            uri = launchUri,
+            token = request.orderId,
+            tokenType = TokenType.ORDER_ID,
+            returnUrlScheme = urlScheme,
+            appLinkUrl = request.appLinkUrl
+        )
+
+        when (result) {
+            is PayPalPresentAuthChallengeResult.Success -> {
+                analytics.notify(
+                    CheckoutEvent.AUTH_CHALLENGE_PRESENTATION_SUCCEEDED,
+                    checkoutOrderId
+                )
+
+                // update auth state value in session store
+                sessionStore.authState = result.authState
+            }
+
+            is PayPalPresentAuthChallengeResult.Failure -> {
+                analytics.notify(
+                    CheckoutEvent.AUTH_CHALLENGE_PRESENTATION_FAILED,
+                    checkoutOrderId
+                )
+            }
+        }
+        return result
+    }
+
+    /**
+     * Confirm PayPal payment source for an order.
+     *
+     * @param request [PayPalWebCheckoutRequest] for requesting an order approval
+     */
+    suspend fun startAsync(
```

Comment:
```suggestion
   fun startAsync(
```
We should wait for `v3` to introduce coroutine APIs. Or even if we have them in `v2`, we should create coroutine APIs for all of our modules.

Reply:
we still support deprecated `fun start`, so added this as new functionality, we can make this private and just keep `fun start(activity, orderRequest, callback)` as the only way to get app switch functionality and add `suspend fun start` in v3. 
Also what other places we can add coroutines support

Reply:
Theoretically `CardClient` and `FraudDetection` might be two modules where we can add coroutine support. It's better if we have a plan for coroutine support first.